raiden/network/proxies/secret_registry.py

```diff
@@ -35,9 +40,11 @@ def __init__(
             to_normalized_address(secret_registry_address),
         )
 
+        # There should be only one smart contract deployed, to avoid race
+        # conditions for on-chain unlocks.
         compare_contract_versions(
             proxy=proxy,
-            expected_version=contract_manager.contracts_version,
+            expected_version='0.4.0',
```

Comment:
Hm why this change? We can use different smart contract versions. We also do so in tests. Some tests use the limit ones and some the normal ones. For example smoke tests fail in your PR and it's probably because they required the non-limit contracts as you can see [here](https://github.com/raiden-network/raiden/blob/a48d5d1f70093869fd235020cbe3d1cf4cebd20e/raiden/ui/cli.py#L560).

Reply:
The rationale is: We should only have one secret registry in the mainnet, if we do have more than one registry things will get unnecessarily complicate because of race conditions. The tests should just use the version of the smart contract that is deployed on the mainnet, otherwise we may be testing against a different behavior.

The problem is that new packages of the raiden-contracts are bumping the version of the smart contracts even if there was no change, so in theory the only different of `0.4.0` to the `0.8.0` is the version number.

I will revert this change, but IMO we should never deploy a new secret registry on the mainnet, maybe the check can be done on a per chain id bases together by the contract address.